### ISSUE TITLE
Watched searches built on other watched searches do not work

### ISSUE THREAD

**Issue by camerojo** [talentbeyondboundaries/tbbtalentv2#143]
See TBBTalent alerts - bug with watched searches

eg Watcher notification failure (Australia - Afghan watch): org.hibernate.LazyInitializationException: failed to lazily initialize a collection of role: org.tbbtalent.server.model.db.SavedSearch.searchJoins, could not initialize proxy - no Session 

This has been around for a while - so not super urgent but it does need to be fixed.

The solution is around loading the base watched search into the JPA session

**Comment by sadatmalik** [comment]
Trying to reproduce this error.

1. Start with a watched search "My Local Testing List 31 Aug 2023"

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL0dkQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c1237517489e031586037de41a5204c61edf8e96/Screenshot%202023-11-03%20at%2010.41.34.png)

2. Create a new watched search - "My Local Testing Search 2 Nov 2023"

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL0tkQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1a61024882d18f34fd796e3af094adffec38a844/Screenshot%202023-11-03%20at%2010.42.41.png)

3. Ensure the second watched search is based on the first watched search:

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL2FkQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--96549c1e5d30f6438fd2a67e2eda485beac09cbf/Screenshot%202023-11-03%20at%2010.43.53.png)

4. Trigger a call to noftify search watchers by temporarily exposing a system api call to savedSearchService.notifySearchWatchers() - log output does not show the exception:

```
2023-11-03 10:47:14.400  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Notify watchers: running 5 searches
2023-11-03 10:47:14.402  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Running search 1: Finance professional
2023-11-03 10:47:14.472  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Found 38 candidates in search
2023-11-03 10:47:14.472  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Running search 2: Human resource manager
2023-11-03 10:47:14.505  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Found 0 candidates in search
2023-11-03 10:47:14.506  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Running search 3: Big 4
2023-11-03 10:47:14.586  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Elasticsearch query:
{
  "bool" : {
    "must" : [
      {
        "simple_query_string" : {
          "query" : "ey ",
          "flags" : -1,
          "default_operator" : "or",
          "analyze_wildcard" : false,
          "auto_generate_synonyms_phrase_query" : true,
          "fuzzy_prefix_length" : 0,
          "fuzzy_max_expansions" : 50,
          "fuzzy_transpositions" : true,
          "boost" : 1.0
        }
      }
    ],
    "filter" : [
      {
        "terms" : {
          "partner.keyword" : [
            "TBB",
            "HIAS",
            "IOM",
            "RP",
            "Dignity",
            "AT",
            "CRS"
          ],
          "boost" : 1.0
        }
      },
      {
        "terms" : {
          "status.keyword" : [
            "active",
            "incomplete",
            "pending",
            "unreachable"
          ],
          "boost" : 1.0
        }
      }
    ],
    "adjust_pure_negative" : true,
    "boost" : 1.0
  }
}
2023-11-03 10:47:14.587  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Elasticsearch sort:
Page request [number: 0, size 1, sort: masterId: DESC,masterId: DESC]
2023-11-03 10:47:14.626  WARN 99085 --- [nio-8080-exec-5] org.elasticsearch.client.RestClient      : request [POST http://localhost:9200/candidates/_search?typed_keys=true&max_concurrent_shard_requests=5&search_type=query_then_fetch&batched_reduce_size=512] returned 1 warnings: [299 Elasticsearch-7.17.11-eeedb98c60326ea3d46caef960fb4c77958fb885 "Elasticsearch built-in security features are not enabled. Without authentication, your cluster could be accessible to anyone. See https://www.elastic.co/guide/en/elasticsearch/reference/7.17/security-minimal-setup.html to enable security."]
2023-11-03 10:47:14.676  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Found 2 candidates in search
2023-11-03 10:47:14.676  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Running search 4: My Local Testing List 31 Aug 2023
2023-11-03 10:47:14.794  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Found 1962 candidates in search
2023-11-03 10:47:14.794  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Running search 5: My Local Testing Search 2 Nov 2023
2023-11-03 10:47:14.875  INFO 99085 --- [nio-8080-exec-5] o.t.s.s.db.impl.SavedSearchServiceImpl   : Found 64 candidates in search

```

**Comment by sadatmalik** [comment]
Stashing changes

**Comment by sadatmalik** [comment]
May have been resolved by #432

**Comment by sadatmalik** [comment]
Is resolved by #432
